src/test/java/com/cappella/csv/CsvClientTests.java

```diff
@@ -0,0 +1,165 @@
+                InputStream is = classloader.getResourceAsStream("missingHeaderTaskName.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_MISSING_GRANT_TASK_NAME_HEADER));
+
+        }
+
+        @Test
+        void testMalformedTaskDueDateCsvFile() {
+                ClassLoader classloader = this.getClass().getClassLoader();
+                InputStream is = classloader.getResourceAsStream("malformedTaskDueDate.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_PARSING_TASK_DUE_DATE));
+        }
+
+        @Test
+        void testMalformedSubTaskDueDateCsvFile() {
+                ClassLoader classloader = this.getClass().getClassLoader();
+                InputStream is = classloader.getResourceAsStream("malformedSubTaskDueDate.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_PARSING_SUB_TASK_DUE_DATE));
+        }
+
+        @Test
+        void testSpecialCharactersCsvFile() {
+        }
+
+        @Test
+        void testOnlyNameCsvFile() {
+        }
+
+        @Test
+        void testSimpleCsvFile() {
+                ClassLoader classloader = this.getClass().getClassLoader();
+                InputStream is = classloader.getResourceAsStream("simple.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().isEmpty());
+                // should also assert that it successfully parsed the grant tasks expected
+        }
+
+        @Test
+        void testProductionCsvFile() {
+        }
+
+        @Test
+        void testParseGrantDueDateWrongFormat() {
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                String dateValue = "1/1/2001";
+                LocalDate dueDate = csv.parseGrantDueDate(dateValue, problems);
+                Assertions.assertNull(dueDate);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_PARSING_TASK_DUE_DATE));
+        }
+
+        @Test
+        void testParseGrantDueDate() {
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                String dateValue = "Jan 1, 2001";
```

Comment:
These three lines of code can be collapsed into a single line of code like so:

LocalDate.of(2001, 01, 01);

Reply:
I created the `dateValue` on purpose because I want to create a LocalDate with it, and I want to pass it in as a String to the method I'm testing on CsvClient (`parseGrantDueDate`).  It is a convenience that I use the same dateformatter that I know the client is using.  I want to use a single variable to calibrate the date I pass to the method and the way I validate the date returned from the method so I don't have the human error of accidentally using two different dates.

Reply:
Right-o.